classes/loginflow/authcode.php

```diff
@@ -642,16 +658,28 @@ protected function handlelogin(string $oidcuniqid, array $authparams, array $tok
                     throw new moodle_exception('errorauthloginfailednouser', 'auth_oidc', null, null, '1');
                 }
             }
-            if (!$CFG->allowaccountssameemail && array_key_exists('email', $userinfo)) {
-                // Here we know there is atleast 1 account with that email.
-                // We can just get the first match which is safe as long as allowaccountssameemail is off.
-                $founduser = $DB->get_record('user', ['email' => $userinfo['email']]);
-                // Before proceeding, set the user to OIDC if they aren't already.
-                $founduser->auth = 'oidc';
-                user_update_user($founduser, false);
-                // Now update the token to match the found username.
-                $DB->set_field('auth_oidc_token', 'username', $founduser->username, ['oidcuniqid' => $username]);
-                $username = $founduser->username;
+            // Ensure user objects are in the right shape for future matching before proceeding.
+            switch ($matchingmethod) {
```

Comment:
Same as above a default catch (could be exception, or a default) would be ideal

Reply:
There is nothing to be done for the default in this case. username matching requires no additional actions